### PR TITLE
Include input & option descriptions as data in registry

### PR DESCRIPTION
**Pull request by eemeli**
One rather important consumer of the registry is syntax highlighting. For that use, we should include the descriptions of the inputs and options in the actual registry data, rather than just in the comments. And the element bodies are a rather excellent place for that.

### CODE REVIEW

File: spec/registry.xml [hunk 1]
```diff
@@ -15,21 +15,25 @@
...
-         "America/New_York".
-      -->
-      <option name="timeZone" validationRule="timeZoneId"/>
+      <option name="dateStyle" values="full long medium short">
+        The predefined date formatting style to use.
+      </option>
+      <option name="timeStyle" values="full long medium short">
+        The predefined time formatting style to use.
+      </option>
+      <option name="calendar" values="buddhist chinese coptic dangi ethioaa ethiopic gregory hebrew indian islamic islamic-umalqura islamic-tbla islamic-civil islamic-rgsa iso8601 japanese persian roc">
```
Comment (aphillips) [inline spec/registry.xml:24]: Which source did you use for these? We should have a pointer as this list is not likely to be stabilized.

File: spec/registry.xml [hunk 2]
```diff
@@ -15,21 +15,25 @@
...
+        Calendar to use.
+      </option>
+      <option name="numberingSystem" values="arab arabext bali beng deva fullwide gujr guru hanidec khmr knda laoo latn limb mlym mong mymr orya tamldec telu thai tibt">
```
Comment (aphillips) [inline spec/registry.xml:27]: This list is incomplete? Can we point to CLDR data here?

**Review by aphillips — APPROVED** [review]
I like the general direction. Some comments (mainly editorial).

File: spec/registry.xml (hunk 1 above)
Comment (eemeli) [reply spec/registry.xml:24]: I only moved the description into the body here; the values were added by @mihnita in #420; I presume that their source is the ECMA-402 spec, which may itself refer to another source.

File: spec/registry.xml (hunk 2 above)
Comment (aphillips) [reply spec/registry.xml:27]: ibid

File: spec/registry.xml [hunk 3]
```diff
@@ -83,63 +102,71 @@
...
+      <option name="compactDisplay" values="short long" default="short">
+          Only used when notation is "compact".
+      </option>
+      <option name="currency" validationRule="currencyCode">
+        The currency to use in currency formatting.
         Possible values are the ISO 4217 currency codes, such as "USD" for the US dollar,
         "EUR" for the euro, or "CNY" for the Chinese RMB — see the
-        Current currency & funds code list
+        Current currency &amp; funds code list
         (https://www.six-group.com/en/products-services/financial-information/data-standards.html#scrollTo=currency-codes).
```
[the remark this answers is not in the document]
Comment (aphillips) [reply spec/registry.xml:113]: CLDR would almost certainly be a better reference for a CLDR-TC subcommittee/working group to use?

Actually, probably we should point to supplmentalData.xml in the "latest" CLDR release for this. 
For numbering systems we should point to numberingSystems.xml in the latest CLDR release.

[the remark this answers is not in the document]
Comment (eemeli) [reply spec/registry.xml]: Replaced with the suggested CLDR link, which is generated from the latest `supplementalData.xml`.

File: spec/registry.dtd [hunk 4]
```diff
@@ -1,4 +1,7 @@
 <!ELEMENT registry (function|validationRule)*>
+<!ATTLIST registry
+    xml:lang       NMTOKEN  #IMPLIED
+>
```
Comment (eemeli) [inline spec/registry.dtd:4]: I added `xml:lang` as an optional attribute on the top-level `<registry>`, as well as directly on `<input>` and `<option>`, to indicate the language of the included descriptions.

Should this be required, though? Or are there use cases where a `<registry>` could be e.g. embedded in another XML document, where the locale is defined somewhere further up the tree?

**Review by aphillips — APPROVED** [review]
sheep it.

🐑

**Pull request merged**